### ISSUE TITLE
Problem about cond_scale 

### ISSUE THREAD

**Issue by kanydao** [lucidrains/video-diffusion-pytorch#35]
Thanks for the great work, I was wondering what the role of cond_scale is in this code  .  
def forward_with_cond_scale(
        self,
        *args,
        cond_scale=2.,
        **kwargs
    ):
        logits = self.forward(*args, null_cond_prob=0., **kwargs)
        if cond_scale == 1 or not self.has_cond:
            return logits

        null_logits = self.forward(*args, null_cond_prob=1., **kwargs)
        return null_logits + (logits - null_logits) * cond_scale

**Comment by chpk** [comment]
The cond_scale parameter modulates how strongly the conditioning influences the generated output, allowing for a flexible adjustment of the model's behavior based on the desired level of conditioning effect. The method forward_with_cond_scale adjusts the model's predictions based on the value of cond_scale and whether conditioning information (has_cond) is present or not.

Here's a breakdown of its functionality:

**_Initial Forward Pass:_** It first performs a standard forward pass through the model with null_cond_prob set to 0, which means that the conditioning information (if any) is fully considered in generating the initial logits or predictions.

**_Conditional Adjustment:_** If cond_scale is equal to 1 or the model does not have conditioning (has_cond is False), the initial logits are returned unchanged. This implies that either the effect of conditioning is not to be scaled or there is no conditioning information to adjust. If cond_scale is not equal to 1 and there is conditioning information, another forward pass is made with null_cond_prob set to 1. This generates null_logits, representing the model's output without considering the conditioning information.

**_Combining Logits with Scaling:_** The final step involves combining the initial logits and the null logits based on the cond_scale value. Specifically, the difference between the logits and null_logits is scaled by cond_scale, and then added back to null_logits. This operation adjusts the influence of the conditioning information on the final output. A higher cond_scale amplifies the effect of the conditioning, potentially leading to outputs that more strongly reflect the conditioning signal.

**Comment by kanydao** [comment]
@chpk Thank you very much for your detailed explanation.
